x-pack/plugins/observability/server/plugin.ts

```diff
@@ -173,6 +173,9 @@ export class ObservabilityPlugin implements Plugin<ObservabilityPluginSetup> {
 
     const { ruleDataService } = plugins.ruleRegistry;
 
+    const savedObjectTypes = config.compositeSlo.enabled
+      ? [SO_SLO_TYPE, SO_COMPOSITE_SLO_TYPE]
+      : [SO_SLO_TYPE];
```

Comment:
Config-base switch for SO type registration seems fairly dangerous. We can easily imagine edge cases where it would be very feasible to break deployments (e.g enable, disable then upgrade).

Mind giving some context on why you think this is necessary?

Reply:
For the context, we have decided to not release composite SLO has it is now until we further validate the use case it solves for SRE and iterate on the solution. We might have to change a few things later regarding the SO, like its schema and what's index, and it's not very recommended to do so once the type has been registered already, right?

The feature flag is there for not registering the SO type by default, and allowing us to continue iterate on the feature.

Nobody should turn this flag on (except internally), and I would be extremely surprised if a customer turns a flag that is not documented anywhere, for a feature that is not even in tech preview.

Reply:
Ok, thanks for clarifying. In that case I think we're fine